### ISSUE TITLE
Problem loading libSDL2-2.0.so

### ISSUE THREAD

**Issue by brunob** [marcusva/py-sdl2#148]
Hi, i'm trying to make a snapcraft package of an app which use pysdl2 and i'm stumbling on an issue for loading `libSDL2-2.0.so` when `PYSDL2_DLL_PATH` is set.

Context : on the host (ubuntu 18.04) the lib is present in `usr/lib/x86_64-linux-gnu/libSDL2-2.0.so.0` so i set `PYSDL2_DLL_PATH` to `usr/lib/x86_64-linux-gnu/`. The problem is that `_findlib` search only for `.so` files, so this lead to an error `ImportError: could not find any library for SDL2 (PYSDL2_DLL_PATH: :/snap/tauon/x1/usr/lib/x86_64-linux-gnu)`.

If i patch the `pattern`used in `_findlib` https://github.com/marcusva/py-sdl2/blob/master/sdl2/dll.py#L11 to `pattern = "lib%s.so.0"` it load the lib and it works well.

Question : should the pattern be updated as what i did, or should it be expanded to match other digits than 0 ? I can provide a pull request for that if you want.

FTR, here is the issue mentioning the bug on the app repo https://github.com/Taiko2k/TauonMusicBox/issues/207#issuecomment-555779442

**Comment by marcusva** [comment]
Thanks for reporting. This is indeed a bug in the pattern matching, assuming that there's always a ``.so`` symlink.
``lib%s.so.0`` as pattern won't be sufficient, since the additional number after the ``.so`` part might be any digit. The pattern match needs to be changed to something along the lines

``lib{0}.so(\.[0-9]+)?``

which would cover ``.so``, ``.so.0``, ``.so.474``, and so on. There was a minor change to the dll code using Python's newer formatting layout, which may need special consideration, when it comes to the regexp match.

**Comment by brunob** [comment]
Thx for answering :) Would you like a PR or will you push the pattern match update ?

**Comment by a-hurst** [comment]
The problem should be fixed as of this commit: https://github.com/marcusva/py-sdl2/commit/3cc7218ec092e24bdf9f9dc0d4e38affe1e28a84

Basically, I wrote things so that on platforms other than macOS or Windows (i.e. ones that would use .so libraries), it searches in the provided `PYSDL2_DLL_PATH` for any .so files matching the library name with a numeric suffix, sorts them newest to oldest (based on suffix, so the newest version will take precedence), and adds them to the list of libraries to attempt loading.

I tested it out in a fresh Linux Mint VM and it worked fine, but @brunob let me know whether it works for you!
